services/blockstorage/service.go

```diff
@@ -62,12 +63,91 @@ func (s *service) CommitBlock(input *services.CommitBlockInput) (*services.Commi
 	return nil, nil
 }
 
+func (s *service) lookForTransactionsBlockHeader(height primitives.BlockHeight) (*services.GetTransactionsBlockHeaderOutput, error) {
+	txBlock, err := s.persistence.GetTransactionsBlock(height)
```

Comment:
styling, maybe change this to
if x,err := action(); err != nil

Reply:
also, not sure here, but if the GetTranBlock(h) is only used here, maybe change the return object to match this function (a bit lost in the API when looking at it like this)